devito/types/dense.py

```diff
@@ -1270,7 +1270,9 @@ def __init_finalize__(self, *args, **kwargs):
         self.time_dim = kwargs.get('time_dim', self.dimensions[self._time_position])
         self._time_order = kwargs.get('time_order', 1)
         super(TimeFunction, self).__init_finalize__(*args, **kwargs)
-
+        # Dynamically add derivative short-cuts
+        self._fd = generate_fd_shortcuts(self.dimensions, self.space_order,
```

Comment:
that's gonna **re**construct the space derivatives as well? since we also get them via ``super``

also, we need surrounding blank lines here

Reply:
Good point, will rethink it

Reply:
fixed